script.js

```diff
@@ -1,4 +1,398 @@
+// Global variables
+var gameMode = "get player name";
+var shuffledDeck = [];
+var computerCard = [];
+var allPlayersHand = []; // Nested Array. Outer array represents players, inner array holds cards which are objects
+var playerName = [];
+var playerBets = [];
+var playerTurn = 0; // variable to keep track of which player's turn
+var playerScore = [];
+var winner = "";
+var winnerDetermined = false;
+
+// Function to generate deck of cards.
+// Picture cards = 10
+var deckGeneration = function () {
+  var cardDeck = [];
+  var suits = ["diamonds", "clubs", "hearts", "spades"];
+
+  for (var i = 0; i < suits.length; i += 1) {
+    var cardSuit = "";
+    if (suits[i] == "diamonds") {
+      cardSuit = "♦️";
+    } else if (suits[i] == "clubs") {
+      cardSuit = "♣️";
+    } else if (suits[i] == "hearts") {
+      cardSuit = "❤️";
+    } else if (suits[i] == "spades") {
+      cardSuit = "♠️";
+    }
+    var index = 1;
+    while (index < 14) {
+      var cardName = index;
+      var rankIndex = index;
+      if (index == 1) {
+        cardName = "Ace";
+        rankIndex = 11; // Let all Aces be worth 11 points from the start.
+      } else if (index == 11) {
+        cardName = "Jack";
+        rankIndex = 10;
+      } else if (index == 12) {
+        cardName = "Queen";
+        rankIndex = 10;
+      } else if (index == 13) {
+        cardName = "King";
+        rankIndex = 10;
+      }
+      var card = {
+        name: cardName,
+        suit: cardSuit,
+        rank: rankIndex,
+      };
+      cardDeck.push(card);
+      index += 1;
+    }
+  }
+  return cardDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+// Function to give 100 points to each player when the game initialise.
+var assignBankRoll = function () {
+  for (var i = 0; i < playerName.length; i += 1) {
+    playerScore.push(Number(100));
+  }
+};
```

Comment:
Nice work splitting this out. In the future you might want to make this function more flexible in the case that you need to give more/less starting points 

So maybe something like this 
```
var assignBankRoll = function (points) {
  for (var i = 0; i < playerName.length; i += 1) {
    playerScore.push(points);
  }
};
```